graphics/libfive/Portfile

```diff
@@ -0,0 +1,165 @@
+PortGroup           cmake 1.1
+
+# libfive does not have releases; the master branch is recommended.
+# See https://libfive.com/download/
+github.setup        libfive libfive 9aa20520497636659b8d5bda25057631837e998a
+version             20210509
+revision            0
+
+categories          graphics math cad
+platforms           darwin
+license             MPL-2
+maintainers         {gmail.com:szhorvat @szhorvat} openmaintainer
+
+homepage            https://libfive.com/
+
+description         Library for 3D solid modeling
+
+long_description    libfive is a software library and set of tools for solid modeling, \
+                    especially suited for parametric and procedural design. It is \
+                    infrastructure for generative design, mass customization, and \
+                    domain-specific CAD tools.
+
+checksums           rmd160  0c28f01b9ccd296fb046c3f0e3c4b3b8b3b6b3e1 \
+                    sha256  7497c1e2cd0c1427cc0d0720d1b19dc9e8d7451156101fef17cfd1b005413f8a \
+                    size    877936
+
+# Even though eigen3 is header-only, it must be a library dependency because
+# the libfive headers include it.
+depends_lib-append  port:boost \
+                    port:eigen3 \
+                    port:libpng
+
+depends_build-append \
+                    path:bin/pkg-config:pkgconfig
+
+compiler.c_standard 1999
+compiler.cxx_standard \
+                    2017
+
+# These will be overridden by enabled variants
+configure.args-append \
+                    -DBUILD_GUILE_BINDINGS=OFF \
+                    -DBUILD_PYTHON_BINDINGS=OFF \
+                    -DBUILD_STUDIO_APP=OFF
+
+patchfiles          patch-remove-march-native.diff
+
+# Guile binding:
+
+# libfive supports Guile 3.0 as well, but MacPorts currently only includes 2.2
+variant guile description {Build guile bindings} {
+    depends_lib-append      port:guile
+    configure.args-replace  -DBUILD_GUILE_BINDINGS=OFF -DBUILD_GUILE_BINDINGS=ON
+
+    build.env-append        GUILE_AUTO_COMPILE=0
+
+    # The Guile bindings are under GPL-2+
+    if { ! ("GPL-2+" in $license) } {
+        license-append GPL-2+
+    }
+}
+
+# Python bindings:
+
+# The minimum supported Python version is 3.7
+set pythons_suffixes {37 38 39}
+
+set pythons_ports {}
+foreach s ${pythons_suffixes} {
+    lappend pythons_ports python${s}
+}
+
+foreach s ${pythons_suffixes} {
+    set p python${s}
+    set v [string index ${s} 0].[string index ${s} 1]
+    set i [lsearch -exact ${pythons_ports} ${p}]
+    set c [lreplace ${pythons_ports} ${i} ${i}]
+
+    variant ${p} description "Build bindings for Python ${v}" conflicts {*}${c} "
+        depends_lib-append      port:${p}
+        configure.args-replace  -DBUILD_PYTHON_BINDINGS=OFF -DBUILD_PYTHON_BINDINGS=ON
+        pre-patch {
+            # Patch the CMake project to use the selected version of Python
+            reinplace \"s|Python3 COMPONENTS|Python3 ${v} EXACT COMPONENTS|g\" ${worksrcpath}/CMakeLists.txt
```

Comment:
instead of the patch I think one should be able to set `Python3_EXECUTABLE` to the correct Python binary (untested):
 
`configure.args-append -DPython3_EXECUTABLE=${frameworks_dir}/Python.framework/Versions/${v}/bin/python${v}`

Reply:
Thanks! This works. I did originally try to set `Python3_EXECUTABLE`, but it did not work. Apparently, it is important to use the full path you've shown here.